spec/controllers/support/schools/devices/allocation_controller_spec.rb

```diff
@@ -0,0 +1,180 @@
+             laptop_allocation: 50, laptop_cap: 40, laptops_ordered: 10,
+             router_allocation: 5, router_cap: 4, routers_ordered: 1)
+    end
+
+    before { get :edit, params: { school_urn: school.urn, device_type: device_type } }
+
+    context 'when the user is not support' do
+      let(:user) { create(:user) }
+
+      it 'forbid' do
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    it 'responds successfully' do
+      expect(response).to be_successful
+    end
+
+    it 'assigns a form for the school laptop allocation' do
+      expect(assigns[:form].device_type).to eq(:laptop)
+      expect(assigns[:form].raw_allocation).to eq(school.raw_laptop_allocation)
+      expect(assigns[:form].raw_devices_ordered).to eq(school.raw_laptops_ordered)
+      expect(assigns[:form].device_allocation).to eq(school.laptop_allocation)
+      expect(assigns[:form].device_cap).to eq(school.laptop_cap)
+    end
+
+    context 'when device type is set' do
+      let(:device_type) { :router }
+
+      it 'assigns a form for that school device allocation' do
+        expect(assigns[:form].device_type).to eq(:router)
+        expect(assigns[:form].raw_allocation).to eq(school.raw_router_allocation)
+        expect(assigns[:form].raw_devices_ordered).to eq(school.raw_routers_ordered)
+        expect(assigns[:form].device_allocation).to eq(school.router_allocation)
+        expect(assigns[:form].device_cap).to eq(school.router_cap)
+      end
+    end
+  end
+
+  describe '#update' do
+    let(:allocation) { 45 }
+    let(:device_type) { :laptop }
+
+    let(:params) do
+      {
+        device_type: device_type,
+        school_urn: school.urn,
+        support_allocation_form: {
+          allocation: allocation,
+        },
+      }
+    end
+
+    let(:rb) do
+      create(:local_authority,
+             :manages_centrally,
+             :vcap_feature_flag,
+             computacenter_reference: '1000')
+    end
+
+    let!(:school) do
+      create(:school,
+             :centrally_managed,
+             :with_std_device_allocation,
+             :with_coms_device_allocation,
+             order_state: 'can_order',
+             computacenter_reference: '11',
+             responsible_body: rb,
+             laptop_allocation: 50, laptop_cap: 40, laptops_ordered: 10,
+             router_allocation: 5, router_cap: 4, routers_ordered: 1)
+    end
+
+    let(:requests) do
+      [
+        [
+          { 'capType' => 'DfE_RemainThresholdQty|Std_Device', 'shipTo' => '11', 'capAmount' => '45' },
+        ],
+      ]
+    end
+
+    before do
+      stub_computacenter_outgoing_api_calls
+    end
+
+    context 'when the user is not support' do
+      let(:user) { create(:user) }
+
+      before { patch :update, params: params }
+
+      it 'forbid' do
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'when the allocation assigned is not valid' do
+      let(:allocation) { '9' }
```

Comment:
This looks like a magic number without scrolling back up

Reply:
Ok, fixing it